apps/gateway-ui/src/components/WithdrawCard.tsx

```diff
@@ -115,29 +88,33 @@ export const WithdrawCard = React.memo(function WithdrawCard({
               fontFamily={theme.fonts.body}
               pb='6px'
             >
-              {t('common.amount')}
+              {`${t('common.amount')} ${t('common.sats')}`}
             </Text>
-            <Input
-              height='44px'
-              p='14px'
-              border={`1px solid ${theme.colors.gray[300]}`}
-              bgColor={theme.colors.white}
-              boxShadow={theme.shadows.xs}
-              borderRadius='8px'
-              w='100%'
-              placeholder={t('withdraw-card.amount-placeholder')}
-              // FIXME: this is a hack
-              value={withdrawObject.amount.toString()}
-              onChange={(e) => handleInputChange(e)}
-              name='amount'
-            />
-
+            <NumberInput
+              min={0}
+              max={balanceMsat / 1000}
+              value={amount}
+              onChange={(value) => {
+                value && setAmount(parseInt(value));
+              }}
```

Comment:
I would avoid attempting to parse an int in state, since inputs work only on strings. We should probably keep this as a string and only parse when we're performing numeric operations. This can cause issues with sufficiently large numbers

<img width="247" alt="Screenshot 2023-10-06 at 8 04 13 AM" src="https://github.com/fedimint/ui/assets/649992/b24851dc-dd75-4642-9b07-61ed9d521bda">

Reply:
This is no worse than it currently is though, so it's fine

Reply:
I see! 👌🏿